oci_python_image/integration_test/service_test.py

```diff
@@ -0,0 +1,36 @@
+import os
+import docker
+from testcontainers.core.container import DockerContainer
+
+TAR_PATH = os.getenv("TEST_SRCDIR") + "/_main/hello_world/tarball/tarball.tar"
+IMAGE_NAME = "gcr.io/oci_python_hello_world:latest"
+
+
+def _load_latest_tarball():
+    """
+    testcontainers requires the image already loaded in the daemon, but rules_oci places
+    a tar file in bazel-out.
+    So, we load the latest tarball to Docker
+    So that we run the test against the latest image
+    """
+    client = docker.from_env()
+    with open(TAR_PATH, "rb") as f:
+        client.images.load(f)
+
+
+def test_container_runs():
+    _load_latest_tarball()
+
+    user = os.environ["USER"]
+
+    with DockerContainer(
+        IMAGE_NAME,
+    ).with_bind_ports(
+        host=9000,
+        container=8080,
+    ) as container:
+        # get_exposed_port waits for the container to be ready
+        # https://github.com/testcontainers/testcontainers-python/blob/2bcb931063e84da1364aa26937778f0e45708000/core/testcontainers/core/container.py#L107-L108
+        port = container.get_exposed_port(8080)
+
+        # TODO(alexeagle): have the application inside the container listen on a port so we can use it as a test fixture
```

Comment:
@thesayyn I'm not sure it's worth landing this until we can actually communicate with a python process running in the container.

Reply:
maybe we could make it a separate example here? and use testcontainers instead of structure_test for testing.

Reply:
I could imagine a separate "integration testing" example that wants to tie together the other stuff in this repo - we'd have our first cross-workspace dependencies though.

I'm getting close to writing a blog post about it, is why I'm working in here at all.